here-naksha-lib-handlers/src/main/java/com/here/naksha/lib/handlers/SourceIdHandler.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2017-2023 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+package com.here.naksha.lib.handlers;
+
+import com.here.naksha.lib.core.IEvent;
+import com.here.naksha.lib.core.INaksha;
+import com.here.naksha.lib.core.models.geojson.implementation.XyzFeature;
+import com.here.naksha.lib.core.models.geojson.implementation.XyzProperties;
+import com.here.naksha.lib.core.models.naksha.EventHandler;
+import com.here.naksha.lib.core.models.naksha.EventHandlerProperties;
+import com.here.naksha.lib.core.models.naksha.EventTarget;
+import com.here.naksha.lib.core.models.storage.*;
+import com.here.naksha.lib.core.util.json.JsonSerializable;
+import com.here.naksha.lib.handlers.util.PropertyOperationUtil;
+import java.util.*;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SourceIdHandler extends AbstractEventHandler {
+
+  private static final Logger logger = LoggerFactory.getLogger(SourceIdHandler.class);
+  private static final String TAG_PREFIX = "xyz_source_id_";
+  private static final String NS_COM_HERE_MOM_META = "@ns:com:here:mom:meta";
+  private static final String SOURCE_ID = "sourceId";
+
+  protected @NotNull EventHandler eventHandler;
+  protected @NotNull EventTarget<?> eventTarget;
+  protected @NotNull EventHandlerProperties properties;
+
+  public SourceIdHandler(
+      final @NotNull EventHandler eventHandler,
+      final @NotNull INaksha hub,
+      final @NotNull EventTarget<?> eventTarget) {
+    super(hub);
+    this.eventHandler = eventHandler;
+    this.eventTarget = eventTarget;
+    this.properties = JsonSerializable.convert(eventHandler.getProperties(), EventHandlerProperties.class);
+  }
+
+  @Override
+  public @NotNull Result processEvent(@NotNull IEvent event) {
```

Comment:
Nicely modularized code overall.
However, I would like to suggest that we avoid using Optional as standard practice, as in my view, for very basic null check also it ends up creating a new object which is unnecessary overhead. It improves a readability a bit, but with an unwanted compromise.